### PR TITLE
pass environment variables to vagrant process

### LINKED ISSUE #36: Support passing environment variables
I need `python-vagrant` to pass environment variables down to the vagrant subprocess call so the code in `Vagrantfile` can act accordingly but this is not possible given the way things are currently set up.

For example, `Vagrantfile` can check in the system's environment if the variable `USE_NFS=1` is set in which case `Vagrantfile` will mount the shared directories using NFS (instead of the default vboxfs). The python tool that wraps `python-vagrant` would know whether this flag should be set to 1 or 0 but it can't currently pass it to `vagrant.up()`.

Would this be something considered if I were to submit a pull request?

### PR DESCRIPTION
**Pull request by aconrad**
Fixes #36.

### CODE REVIEW

**Comment by sontek** [comment]
@aconrad Would you mind taking a look at the tests directory and trying to add a new test that verifies this change?

Something like: https://github.com/todddeluca/python-vagrant/blob/master/tests/test_vagrant.py#L377-L392

**Comment by sontek** [comment]
@aconrad Doesn't merge cleanly anymore either

**Comment by aconrad** [comment]
@todddeluca I've updated the CHANGES and the README. I still need to work on the tests.

I was wondering if it was useful to even introduce the `env` kwarg of `Vagrant(env=env)`. Users may just assign `Vagrant.env` directly. Passing `env` as part of `__init__` is just for convenience but I don't feel it's actually useful (and more code to maintain). We could add it later if there is a demand for it. What do you think?

File: vagrant/__init__.py [hunk 1]
```diff
@@ -210,6 +213,7 @@ def up(self, no_provision=False, provider=None, vm_name=None,
            provision=None, provision_with=None):
         '''
         Launch the Vagrant box.
+        vm_name=None: name of VM.
```
Comment (aconrad) [inline vagrant/__init__.py]: Note: this fix is out of scope.

File: vagrant/__init__.py [hunk 2]
```diff
@@ -164,7 +164,7 @@ class Vagrant(object):
         'ubuntu-precise64': 'http://files.vagrantup.com/precise64.box',
     }
 
-    def __init__(self, root=None, quiet_stdout=True, quiet_stderr=True):
+    def __init__(self, root=None, quiet_stdout=True, quiet_stderr=True, env=None):
```
Comment (aconrad) [inline vagrant/__init__.py]: I wonder if `env=None` is very useful to add. Maybe we can just let people fiddle with `v.env` directly.

**Comment by todddeluca** [comment]
One could consider passing `env` via the constructor a matter of style. I prefer to include the environment in the constructor, since doing so:

- documents the "parameters" of the object, letting a user know that they can construct a Vagrant instance with a non-default environment.
- offers the convenience of setting the environment while constructing the object.
- maintains the illusion that a `Vagrant` instance is an immutable object, for the functionally inclined among us.

Of course, if someone needs a different environment for a specific method call, they can tweak the `env` member directly (or create a second Vagrant instance).
